### PR TITLE
langchain[minor]: use langsmith sdk for prompts functionality instead of langchain hub

### PR DESCRIPTION
**Pull request by madams0013**

### CODE REVIEW

**Review by bracesproul — APPROVED** [review]
lgtm

File: langchain/src/hub.ts [hunk 1]
```diff
@@ -13,10 +13,26 @@ import { load } from "./load/index.js";
 export async function push(
   repoFullName: string,
   runnable: Runnable,
-  options?: HubPushOptions & ClientConfiguration
+  options?: {
+    apiKey?: string;
+    apiUrl?: string;
+    parentCommitHash?: string;
+    newRepoIsPublic?: boolean;
```
Comment (jacoblee93) [inline langchain/src/hub.ts:20]: Better thing to do would be to deprecate them like this:

```
/** @deprecated Use isPublic instead. */
newRepoIsPublic?: boolean;
isPublic?: boolean;
```

File: langchain/src/hub.ts [hunk 2]
```diff
@@ -13,10 +13,30 @@ import { load } from "./load/index.js";
...
+    readme?: string;
+    tags?: string[];
+  }
 ) {
   const client = new Client(options);
-  return client.push(repoFullName, JSON.stringify(runnable), options);
+  const payloadOptions = {
+    object: runnable,
+    parentCommitHash: options?.parentCommitHash,
+    isPublic: options?.isPublic,
```
Comment (jacoblee93) [inline langchain/src/hub.ts:34]: Oh, you still need to respect the old values:

```
isPublic: options?.isPublic ?? options?.newRepoIsPublic,
```

File: langchain/src/hub.ts [hunk 3]
```diff
@@ -13,10 +13,30 @@ import { load } from "./load/index.js";
...
+    description: options?.description,
```
Comment (jacoblee93) [inline langchain/src/hub.ts:35]: Same here, see above

**Review by jacoblee93 — APPROVED** [review]

**Comment by jacoblee93** [comment]
Thanks!

**Pull request merged**